### DISCUSSION TITLE
Suspend an app temporarily?

### DISCUSSION THREAD

**Discussion by dbieber** [Textualize/textual#4002] [Q&A]
I'm interested in running $EDITOR (the user's specified editor) or dropping into ipython (IPython.embed()) from within a running textual app. I've seen that [textual-terminal](https://github.com/mitosch/textual-terminal) allows for running subprocesses within a textual app's ui through it's use of [pyte](https://github.com/selectel/pyte/blob/master/examples/terminal_emulator.py), but I also want to explore running these full-screen commands outside of the textual app's UI. That is, I want to temporarily suspend the app, run the subcommand (either the editor or IPython), and then resume the textual app after that. Is that something that textual supports?

@blob42 feel free to weigh in since it looks like you attempted something similar in https://github.com/Textualize/textual/discussions/2741.

**Comment by davep** [comment]
Currently in our TODO list: #3727

**Reply by dbieber** [reply]
Thanks for the pointer. I don't think that's the meaning of "suspend" I meant though. I don't want to suspend the process running the textual app.

Rather I want to clear the app's screen (so it shows the terminal however it was when the user launched the app) and pause rendering, do some stuff of my own with the terminal in its default settings, and then resume rendering.

**Reply by davep** [reply]
I must be misunderstanding then, as you seem to be describing exactly what the `suspend` context manager was designed to do in the PR linked to in the above-mentioned issue.

**Reply by dbieber** [reply]
Oh, my bad! I saw the bits about suspending the _process_ (action_suspend_process) and thought it was the wrong thing. But the other part of the PR - the paused and suspend contextmanagers - are indeed what I was looking for. Ty.

**Comment by blob42** [comment]
@dbieber Yes I implement exactly this feature here https://github.com/blob42/Instrukt/blob/c15cde6366922af3dc250f9b103b3bf56c0cc516/instrukt/app.py#L396C1-L396C1 and subprocess.py
